pyface/undo/undo_manager.py

```diff
@@ -0,0 +1,122 @@
+# ------------------------------------------------------------------------------
+# Copyright (c) 2008, Riverbank Computing Limited
+# All rights reserved.
+#
+# This software is provided without warranty under the terms of the BSD
+# license included in enthought/LICENSE.txt and may be redistributed only
+# under the conditions described in the aforementioned license.  The license
+# is also available online at http://www.enthought.com/licenses/BSD.txt
+# Thanks for using Enthought open source!
+#
+# Author: Riverbank Computing Limited
+# Description: <Enthought undo package component>
+# ------------------------------------------------------------------------------
+
+# Enthought library imports.
+from traits.api import (
+    Bool,
+    Event,
+    HasTraits,
+    Instance,
+    Int,
+    Property,
+    Str,
+    provides,
+)
+
+# Local imports.
+from .i_undo_manager import IUndoManager
+
+
+@provides(IUndoManager)
+class UndoManager(HasTraits):
+    """The UndoManager class is the default implementation of the
+    IUndoManager interface.
+    """
+
+    #### 'IUndoManager' interface #############################################
+
+    # This is the currently active command stack and may be None.  Typically it
+    # is set when some sort of editor becomes active.
+    active_stack = Instance("apptools.undo.api.ICommandStack")
```

Comment:
not sure why we can't just import and use `ICommandStack` here instead of using the string.

Reply:
There is a circular import, I think - `UndoManager` references `ICommandStack` and vice-versa.